portability-gateway/src/main/java/org/dataportabilityproject/gateway/reference/OauthCallbackHandler.java

```diff
@@ -0,0 +1,186 @@
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dataportabilityproject.gateway.reference;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.io.BaseEncoding;
+import com.google.common.net.HttpHeaders;
+import com.google.inject.Inject;
+import com.sun.net.httpserver.Headers;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import java.io.IOException;
+import java.util.Map;
+import javax.crypto.SecretKey;
+import org.dataportabilityproject.gateway.ApiFlags;
+import org.dataportabilityproject.gateway.crypto.DecrypterFactory;
+import org.dataportabilityproject.gateway.crypto.EncrypterFactory;
+import org.dataportabilityproject.gateway.crypto.SymmetricKeyGenerator;
+import org.dataportabilityproject.gateway.reference.ReferenceApiUtils.FrontendConstantUrls;
+import org.dataportabilityproject.gateway.reference.ReferenceApiUtils.HttpMethods;
+import org.dataportabilityproject.gateway.reference.ReferenceApiUtils.TransferMode;
+import org.dataportabilityproject.spi.cloud.storage.JobStore;
+import org.dataportabilityproject.spi.cloud.types.PortabilityJob;
+import org.dataportabilityproject.spi.cloud.types.TypeManager;
+import org.dataportabilityproject.spi.gateway.auth.AuthDataGenerator;
+import org.dataportabilityproject.spi.gateway.auth.AuthServiceProvider;
+import org.dataportabilityproject.spi.gateway.auth.AuthServiceProviderRegistry;
+import org.dataportabilityproject.types.transfer.auth.AuthData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * HttpHandler for callbacks from Oauth1 authorization flow. Redirects client request to: - the next
+ * authorization (if this is after the source service auth) or - the copy page (if this is after the
+ * destination service auth)
+ */
+final class OauthCallbackHandler implements HttpHandler {
+  private final Logger logger = LoggerFactory.getLogger(OauthCallbackHandler.class);
+  public static final String PATH = "/callback1/";
+
+  // TODO: obtain from flags
+  private static final boolean IS_LOCAL = true;
+
+
+  private final AuthServiceProviderRegistry registry;
+  private final JobStore store;
+  private final SymmetricKeyGenerator symmetricKeyGenerator;
+  private final ObjectMapper objectMapper;
+
+  @Inject
+  OauthCallbackHandler(
+      JobStore store,
+      AuthServiceProviderRegistry registry,
+      SymmetricKeyGenerator symmetricKeyGenerator,
+      TypeManager typeManager
+  ) {
```

Comment:
formatting question, is it usual for the ) { to go on the next line when the parameter list is split up like this? I havent seen this style before.

Reply:
IntelliJ is formatting that way for me, we can try to formalize our style guide a bit more and setup intellij to work that way. Maybe we can discuss in the slack?

Reply:
+1, I agree it would be nice to have a more consistent style guide. If we choose to enforce Google standard, there is an intelliJ extension we could all install: https://github.com/google/google-java-format 

I can also take a look at how to have a linter run as part of travis or something else on github.